[IValue] Better handle sparseTensors in extractStorages

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100783

Sparse tensors don't seem to be handled when we have tensors instead
of pyobjects.

Differential Revision: [D45632427](https://our.internmc.facebook.com/intern/diff/D45632427/)

rohan-varma committed:
[IValue] Better handle sparseTensors in extractStorages

Sparse tensors don't seem to be handled when we have tensors instead
of pyobjects.

Differential Revision: [D45632427](https://our.internmc.facebook.com/intern/diff/D45632427/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[IValue] Better handle sparseTensors in extractStorages"


Sparse tensors don't seem to be handled when we have tensors instead
of pyobjects.

Differential Revision: [D45632427](https://our.internmc.facebook.com/intern/diff/D45632427/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[IValue] Better handle sparseTensors in extractStorages"


Sparse tensors don't seem to be handled when we have tensors instead
of pyobjects.

Differential Revision: [D45632427](https://our.internmc.facebook.com/intern/diff/D45632427/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[IValue] Better handle sparseTensors in extractStorages"


Sparse tensors don't seem to be handled when we have tensors instead
of pyobjects.

Differential Revision: [D45632427](https://our.internmc.facebook.com/intern/diff/D45632427/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[IValue] Better handle sparseTensors in extractStorages"


Sparse tensors don't seem to be handled when we have tensors instead
of pyobjects.

Differential Revision: [D45632427](https://our.internmc.facebook.com/intern/diff/D45632427/)

[ghstack-poisoned]

H-Huang commented on test/cpp/jit/test_misc.cpp:
```diff
@@ -2399,6 +2399,22 @@ TEST(FuturesTest, Basic) {
+  }
+  auto f = c10::make_intrusive<Future>(TensorType::get());
+  at::TensorOptions opts = at::TensorOptions().device(at::DeviceType::CUDA);
+  auto sparse_tensor = at::sparse_coo_tensor(torch::ones(10), torch::ones(10), opts);
```
The following should work for creating a sparse tensor:

`at::sparse_coo_tensor(torch::ones(1, 10), torch::ones(10, 10), opts);`

H-Huang replied:
Would also recommend creating a dense tensor and calling `.to_sparse()` as that will give the indices a different representation

rohan-varma replied:
thanks will try!

H-Huang reviewed (approved):
Just need to fix test. Stamp to unblock

rohan-varma committed:
Update on "[IValue] Better handle sparseTensors in extractStorages"


Sparse tensors don't seem to be handled when we have tensors instead
of pyobjects.

Differential Revision: [D45632427](https://our.internmc.facebook.com/intern/diff/D45632427/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[IValue] Better handle sparseTensors in extractStorages"


Sparse tensors don't seem to be handled when we have tensors instead
of pyobjects.

Differential Revision: [D45632427](https://our.internmc.facebook.com/intern/diff/D45632427/)

[ghstack-poisoned]

rohan-varma commented:
@pytorchbot merge -f "CI done"
